src/index.js

```diff
@@ -3,16 +3,21 @@ import PropTypes from 'prop-types';
 
 import Message from './components/message';
 
-class NetworkStateNotifier extends React.Component {
-  constructor(props) {
-    super(props);
+const inBrowser = typeof navigator !== 'undefined';
+const unsupportedUserAgentsPattern = /Windows.*Chrome|Windows.*Firefox|Linux.*Chrome/;
+const pollingURL = 'https://ipv4.icanhazip.com/';
 
-    this.state = {
-      checker: null,
-      isOnline: navigator.onLine,
-      messages: [],
-    };
-  }
+class NetworkStateNotifier extends React.Component {
+  state = {
+    checker: null,
+    isOnline:
+      inBrowser && typeof navigator.onLine === 'boolean'
+        ? navigator.onLine
+        : true,
```

Comment:
setting the default value to `true`?

Reply:
This case arises when the user is probably running a script and not in a browser and there no property for navigator object/it has been manipulated. It was a confusing decision.

Reply:
You can set it to undefined. which is again a false value. I don't think setting to true is good.

Reply:
I will set it undefined and run the tests. But wouldn't setting false be the same?

Reply:
false is fine too. But the point is you don't the value of isOnline exactly. So I suggested undefined.